File: src/data/courses/dsa/prefix-sum/prefix-sum-fundamentals/continuous-subarray-sum/code/java/ContinuousSubarraySum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        Map<Integer, Integer> modMap = new HashMap<>();
        modMap.put(0, -1); // Initialize with 0 modulo at index -1
        
        int prefixSum = 0;
        
        for (int i = 0; i < nums.length; i++) {
            prefixSum += nums[i];
            int mod = prefixSum % k;
            
            // Check if we've seen this modulo before
            if (modMap.containsKey(mod)) {
                // Check if the subarray length is at least 2
                if (i - modMap.get(mod) >= 2) {
                    return true;
                }
            } else {
                // Store first occurrence of this modulo
                modMap.put(mod, i);
            }
        }
        
        return false;
    }
}

File: src/data/courses/dsa/prefix-sum/prefix-sum-fundamentals/continuous-subarray-sum/code/python/ContinuousSubarraySum.py
from collections import defaultdict

class Solution:
    def checkSubarraySum(self, nums: list[int], k: int) -> bool:
        mod_map = defaultdict(lambda: -1)
        mod_map[0] = -1  # Initialize with 0 modulo at index -1
        
        prefix_sum = 0
        
        for i in range(len(nums)):
            prefix_sum += nums[i]
            mod = prefix_sum % k
            
            # Check if we've seen this modulo before
            if mod in mod_map:
                # Check if the subarray length is at least 2
                if i - mod_map[mod] >= 2:
                    return True
            else:
                # Store first occurrence of this modulo
                mod_map[mod] = i
        
        return False
